api/src/main/kotlin/team/backend/common/logging/decorator/LoggingRequestDecorator.kt

```diff
@@ -0,0 +1,41 @@
+package team.backend.common.logging.decorator
+
+import org.slf4j.Logger
+import org.springframework.core.io.buffer.DataBuffer
+import org.springframework.http.HttpMethod
+import org.springframework.http.server.reactive.ServerHttpRequest
+import org.springframework.http.server.reactive.ServerHttpRequestDecorator
+import org.springframework.util.StringUtils
+import reactor.core.publisher.Flux
+import team.backend.support.asString
+import java.io.ByteArrayOutputStream
+import java.nio.channels.Channels
+import java.util.*
+
+class LoggingRequestDecorator internal constructor(log: Logger, delegate: ServerHttpRequest): ServerHttpRequestDecorator(delegate) {
+
+    private val body: Flux<DataBuffer>?
+
+    override fun getBody(): Flux<DataBuffer> {
+        return body!!
+    }
+
+    init {
+        if (log.isInfoEnabled) {
+            val path = delegate.uri.path
+            val query = delegate.uri.query
+            val method = Optional.ofNullable(delegate.method).orElse(HttpMethod.GET).name()
+            val headers = delegate.headers.asString()
+            log.info(
+                "{} {}\n {}", method, path + (if (StringUtils.hasText(query)) "?$query" else ""), headers
+            )
+            body = super.getBody().doOnNext { buffer: DataBuffer ->
+                val bodyStream = ByteArrayOutputStream()
+                Channels.newChannel(bodyStream).write(buffer.asByteBuffer().asReadOnlyBuffer())
+                log.info("{}: {}", ">>> Request", String(bodyStream.toByteArray()))
```

Comment:
You didn't set anything else, but is there a \n front and back?

Reply:
I didn't understand the intent of the question.

Reply:
@BAEKJungHo 
If you look at the request, response you mentioned in the text, \n is attached front and back.
But I don't see where to set it up.

Reply:
@mangchhe \n is not conventions just samples. It's because the json format had a new line at the time of the request.